# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/brush_mix_blend.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/brush_mix_blend.glsl
--- a/gfx/wr/webrender/res/brush_mix_blend.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/brush_mix_blend.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -8,18 +8,18 @@
 #include shared,prim_shared,brush
 
 // UV and bounds for the source image
-varying vec2 v_src_uv;
-flat varying vec4 v_src_uv_sample_bounds;
+varying highp vec2 v_src_uv;
+flat varying highp vec4 v_src_uv_sample_bounds;
 
 // UV and bounds for the backdrop image
-varying vec2 v_backdrop_uv;
-flat varying vec4 v_backdrop_uv_sample_bounds;
+varying highp vec2 v_backdrop_uv;
+flat varying highp vec4 v_backdrop_uv_sample_bounds;
 
 // Flag to allow perspective interpolation of UV.
 // Packed in to vector to work around bug 1630356.
-flat varying vec2 v_perspective;
+flat varying mediump vec2 v_perspective;
 // mix-blend op. Packed in to vector to work around bug 1630356.
-flat varying ivec2 v_op;
+flat varying mediump ivec2 v_op;
 
 #ifdef WR_VERTEX_SHADER
 